src/main/java/graphql/schema/PropertyFetchingImpl.java

```diff
@@ -179,6 +197,28 @@ private Method findPubliclyAccessibleMethod(CacheKey cacheKey, Class<?> rootClas
         return rootClass.getMethod(methodName);
     }
 
+    /*
+       https://docs.oracle.com/en/java/javase/15/language/records.html
+
+       A record class declares a sequence of fields, and then the appropriate accessors, constructors, equals, hashCode, and toString methods are created automatically.
+
+       Records cannot extend any class - so we need only check the root class for a publicly declared method with the propertyName
+     */
+    private Method findRecordMethod(CacheKey cacheKey, Class<?> rootClass, String methodName) throws NoSuchMethodException {
+        if (Modifier.isPublic(rootClass.getModifiers())) {
```

Comment:
Records are also `final`, this should be restricted to records or other `final` classes.

```if (Modifier.isPublic(rootClass.getModifiers()) && Modifier.isFinal(rootClass.getModifiers())) {```

Otherwise could end up in an odd situation where non-final classes only have root class record-ish methods added.